src/pyggdrasil/tree_inference/_tree_generator.py

```diff
@@ -0,0 +1,127 @@
+"""This submodule defines functions to generate trees.
+
+All adjacency matrices are assumed to have the highest index node as the root.
+All Adjacency matrices are assumed to be directed, and to have no self-connections.
+
+Note: this is part of the private API.
+"""
+import jax
+
+import numpy as np
+
+from pyggdrasil.tree_inference import JAXRandomKey
+
+
+def generate_star_tree(n_nodes: int) -> np.ndarray:
+    """Generate a star tree of n_nodes nodes
+       Root is the highest index node.
+
+    Args:
+        n_nodes: Number of nodes, including root, in the tree.
+
+    Returns:
+        Adjacency matrix of a star tree with n_nodes nodes.
+        A tree with n_nodes nodes, each node descending from the root.
+        I.e. matrix with last row all ones, but the last as root.
+        Nodes are not self-connected.
+    """
+    tree = np.zeros((n_nodes, n_nodes), dtype=int)
+    tree[-1, :] = 1
+    tree[-1, -1] = 0  # root is not self-connected
+
+    return tree
+
+
+def generate_deep_tree(rng: JAXRandomKey, n_nodes: int) -> np.ndarray:
+    """Generate a deep tree of n_nodes nodes.
+
+    Args:
+        rng: JAX random key.
+        n_nodes: Number of nodes in the tree.
+
+    Returns:
+        Adjacency matrix of a deep tree with n_nodes nodes.
+        Root is the highest index node.
+        Nodes are not self-connected.
+    """
+    tree = np.zeros((n_nodes, n_nodes), dtype=int)
+
+    # generate a random permutation of the nodes
+    nodes_except_root = n_nodes - 1
+    permutation = jax.random.permutation(rng, nodes_except_root)
```

Comment:
This is smart – I wanted to ask what happens if the permutation has a fixed point, e.g., $p(3) = 3$, but it's not really a problem due to the implementation :slightly_smiling_face: